### ISSUE TITLE
Support the decode_responses argument

### ISSUE THREAD

**Issue by twidi** [jamesls/fakeredis#77]
In `redis-py`, `StrictRedis` accepts a `decode_responses` argument, set to `False` by default, but when set to `True`, it will decode all bytes values using its `charset` argument, default to utf-8.

When my team uses Redis, we set this argument to `True`. So we expect strings (we use python 3), not bytes.
But `fakeredis` always returns bytes. It encodes everything in bytes but never decodes.

So in our tests we have inconsistent behavior if we swap between a real redis server and `fakeredis`.

Viewing the code of `fakeredis` it's not so easy to support it because not everything goes through a single method, as does `redis-py` with `read_response`.

Do you think it's possible to have this feature? Would you accept a pull-request on this subject?

**Comment by twidi** [comment]
By the way, we use a layer on top of redis/fakeredis, so we have a method through which every response go, so I was able to update it this way:

```python
def decode_result(self, result):
    encoding = self.connection._encoding  # FakeStrictRedis

    def dec_bytes(string):
        return string.decode(encoding)

    def dec_iter(val):
        return val.__class__((dec(v) for v in val))

    def dec_items(val):
        return val.__class__(((dec(k), dec(v)) for k, v in val.items()))

    def dec(val):

        # Decode bytes.
        if isinstance(val, bytes):
            return dec_bytes(val)

        # Decode dicts or dict-like objects
        if hasattr(val, 'items'):
            return dec_items(val)

        # Decode other iterable, but not strings.
        if not isinstance(val, str) and isinstance(val, Iterable):
            return dec_iter(val)

        # No decoding for int, float, str
        return val

    return dec(result)
```

Not sure yet if it will handle all the cases, but maybe using something like that in every redis-like functions that return something could help.

**Comment by matt-snider** [comment]
This is something I would also like to use

**Comment by SpotlightKid** [comment]
We would also need this for out tests, since we use `redis.StrictRedis` with `decode_responses` in out production code.

**Comment by jamesls** [comment]
Implemented in #123
